scipy/sparse/linalg/isolve/lsqr.py

```diff
@@ -485,10 +486,13 @@ def lsqr(A, b, damp=0.0, atol=1e-8, btol=1e-8, conlim=1e8,
         #    Estimate r1norm from
         #    r1norm = sqrt(r2norm^2 - damp^2*||x||^2).
         # Although there is cancellation, it might be accurate enough.
-        r1sq = rnorm**2 - dampsq * xxnorm
-        r1norm = sqrt(abs(r1sq))
-        if r1sq < 0:
-            r1norm = -r1norm
+        if damp > 0:
+            r1sq = rnorm**2 - dampsq * xxnorm
+            r1norm = sqrt(abs(r1sq))
+            if r1sq < 0:
+                r1norm = -r1norm
+        else:
```

Comment:
Not a criticism, just a note for when someone more comfortable with this module reviews--the cases with `damp > 0` are not currently covered by unit tests so may want to double check:

![image](https://user-images.githubusercontent.com/7903078/111055597-baad0880-8434-11eb-8a5a-cdf748ea174f.png)

Reply:
I took the liberty to add `damp > 0` to the `test_basic`.